src/dbus/mutter_display_config.rs

```diff
@@ -308,7 +306,8 @@ impl DisplayConfig {
         // Sort by connector.
         monitors.sort_unstable_by(|a, b| a.0.names.0.cmp(&b.0.names.0));
 
-        let (monitors, logical_monitors) = monitors.into_iter().unzip();
+        let (monitors, logical_monitors): (Vec<_>, Vec<_>) = monitors.into_iter().unzip();
+        let logical_monitors = logical_monitors.into_iter().flatten().collect();
```

Comment:
Seems like at this point it's better to make two Vecs, then manually `for (a, b) in monitors.into_iter()` and push the items in.

Reply:
Sure.. didn't want to change the existing code too much but it would be better